cmd/cluster-init/generate_job.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+	utilpointer "k8s.io/utils/pointer"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/jobconfig"
+)
+
+const (
+	LatestImage = api.ServiceDomainAPPCIRegistry + "/ci/applyconfig:latest"
+	LabelRole   = "ci.openshift.io/role"
+	Infra       = "infra"
+)
+
+func generatePeriodic(clusterName string) prowconfig.Periodic {
+	return prowconfig.Periodic{
+		JobBase: prowconfig.JobBase{
+			Name:       RepoMetadata().SimpleJobName(jobconfig.PeriodicPrefix, clusterName+"-apply"),
+			Agent:      string(prowapi.KubernetesAgent),
+			Cluster:    string(api.ClusterAPPCI),
+			SourcePath: "",
+			Spec: &v1.PodSpec{
+				Volumes: []v1.Volume{generateSecretVolume(clusterName)},
+				Containers: []v1.Container{
+					generateContainer("applyconfig:latest",
+						clusterName,
+						[]string{"--confirm=true"},
+						[]v1.VolumeMount{})},
```

Comment:
nit: calling code should be able to handle `nil` as well as an explicitly empty slice, and `nil` is IMHO more semantically meaningful - "I have nothing to add" rather than "add this specific empty nothing"